spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/cassandra/CassandraMetricsBinder.java

```diff
@@ -0,0 +1,153 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.metrics.cassandra;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.codahale.metrics.Counting;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import com.datastax.oss.driver.api.core.CqlSession;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+import org.springframework.util.Assert;
+
+/**
+ * This is a {@link MeterBinder} that binds all available Cassandra DataStax driver
+ * metrics to Micrometer.
+ *
+ * @author Erik Merkle
+ * @since 2.3.0
+ */
+public class CassandraMetricsBinder implements MeterBinder {
+
+	/**
+	 * Prefix for all Driver configuration parameters.
+	 */
+	public static final String SESSION_PREFIX = "cassandra.driver.session";
+
+	/**
+	 * Description for micrometer metrics.
+	 */
+	public static final String DESCRIPTION = "Please see reference.conf for more information.";
+
+	private static final Map<String, Function<Snapshot, Number>> TIMER_SNAPSHOT_METRICS = new HashMap<>();
+
+	private static final Map<String, Function<Timer, Number>> TIMER_METRICS = new HashMap<>();
+
+	static {
+		TIMER_SNAPSHOT_METRICS.put("p999", Snapshot::get999thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p99", Snapshot::get99thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p98", Snapshot::get98thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p95", Snapshot::get95thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("p75", Snapshot::get75thPercentile);
+		TIMER_SNAPSHOT_METRICS.put("min", Snapshot::getMin);
+		TIMER_SNAPSHOT_METRICS.put("max", Snapshot::getMax);
+		TIMER_SNAPSHOT_METRICS.put("mean", Snapshot::getMean);
+		TIMER_SNAPSHOT_METRICS.put("median", Snapshot::getMedian);
+		TIMER_SNAPSHOT_METRICS.put("stdDev", Snapshot::getStdDev);
+	}
+
+	static {
+		TIMER_METRICS.put("mean_rate", Timer::getMeanRate);
+		TIMER_METRICS.put("m1_rate", Timer::getOneMinuteRate);
+		TIMER_METRICS.put("m5_rate", Timer::getFiveMinuteRate);
+		TIMER_METRICS.put("m15_rate", Timer::getFifteenMinuteRate);
+	}
```

Comment:
I'd rather wrap these map in an unmodifiable map:

```suggestion
	private static final Map<String, Function<Snapshot, Number>> TIMER_SNAPSHOT_METRICS;

	private static final Map<String, Function<Timer, Number>> TIMER_METRICS;

	static {
		Map<String, Function<Snapshot, Number>> map = new HashMap<>();
		map.put("p999", Snapshot::get999thPercentile);
		map.put("p99", Snapshot::get99thPercentile);
		map.put("p98", Snapshot::get98thPercentile);
		map.put("p95", Snapshot::get95thPercentile);
		map.put("p75", Snapshot::get75thPercentile);
		map.put("min", Snapshot::getMin);
		map.put("max", Snapshot::getMax);
		map.put("mean", Snapshot::getMean);
		map.put("median", Snapshot::getMedian);
		map.put("stdDev", Snapshot::getStdDev);
		TIMER_SNAPSHOT_METRICS = Collections.unmodifiableMap(map);
	}

	static {
		Map<String, Function<Timer, Number>> map = new HashMap<>();
		map.put("mean_rate", Timer::getMeanRate);
		map.put("m1_rate", Timer::getOneMinuteRate);
		map.put("m5_rate", Timer::getFiveMinuteRate);
		map.put("m15_rate", Timer::getFifteenMinuteRate);
		TIMER_METRICS = Collections.unmodifiableMap(map);
	}
```